src/StatusQ/Components/StatusItemSelector.qml

```diff
@@ -0,0 +1,226 @@
+
+   Example of how the component looks like:
+   \image status_item_selector.png
+
+   Example of how to use it:
+   \qml
+        StatusItemSelector {
+            icon: Style.svg("contact_verified")
+            title: qsTr("Who holds")
+            defaultItemText: qsTr("Example: 10 SNT")
+            andOperatorText: qsTr("and")
+            orOperatorText: qsTr("or")
+            popupItem: CustomPopup {
+                id: customPopup
+                onAddItem: {
+                    tokensSelector.addItem(itemText, itemImage, operator)
+                    customPopup.close()
+                }
+            }
+        }
+   \endqml
+   For a list of components available see StatusQ.
+*/
+Rectangle {
+    id: root
+    /*!
+       \qmlproperty string StatusItemSelector::icon
+       This property holds the icon name for the icon represented on top of the component as a title icon.
+    */
+    property string icon
+    /*!
+       \qmlproperty string StatusItemSelector::iconSize
+       This property holds the icon size for the icon represented on top of the component as a title icon.
+    */
+    property int iconSize: 18
+    /*!
+       \qmlproperty string StatusItemSelector::title
+       This property holds the titel shown on top of the component.
+    */
+    property string title
+    /*!
+       \qmlproperty string StatusItemSelector::defaultItemText
+       This property holds the default item text shown when the list of items is empty.
+    */
+    property string defaultItemText
+    /*!
+       \qmlproperty string StatusItemSelector::defaultItemImageSource
+       This property holds the default item icon shown when the list of items is empty.
+    */
+    property url defaultItemImageSource: ""
+    /*!
+       \qmlproperty string StatusItemSelector::popupItem
+       This property holds a custom popup item to be opened near the `add` button in order to select new items.
+    */
+    property QC.Popup popupItem
+    /*!
+       \qmlproperty string StatusItemSelector::itemsModel
+       This property holds the data that will be populated in the items selector.
+
+       Here an example of the model roles expected:
+       \qml
+            itemsModel: ListModel {
+            ListElement {
+                text: "Socks"
+                imageSource: "qrc:imports/assets/png/tokens/SOCKS.png"
+                operator: Utils.Operator.None
+            }
+            ListElement {
+                text: "ZRX"
+                imageSource: "qrc:imports/assets/png/tokens/ZRX.png"
+                operator: Utils.Operator.Or
+            }
+        }
+       \endqml
+    */
+    property ListModel itemsModel: ListModel { }
+    /*!
+       \qmlproperty string StatusItemSelector::andOperatorText
+       This property holds the string text representation for an `AND` logical operator.
+    */
+    property string andOperatorText: "and"
```

Comment:
if those are displayed to the user, it should be marked so, eg. `qsTr("and")`

Reply:
We are not using `qsTr` in `StatusQ`. It is still a discussion in how to approach this but for now in `StatusQ` we are providing just only a default value and exposing the property to be correctly assigned by the consumer.

Reply:
I see... but providing the correct default here would be sufficient in this case

Reply:
To give some context here, we originally didn't use qsTr() in StatusQ because our setup would change the source files from qsTr() to qsTrId().

It didn't feel like to use something like StatusQ and then have its source being changed by some mechanism that's coming from the app.

I believe this is no longer an issue though (would have to verify with @osmaczko and @jrainville on what's the latest state).

We should definitely move to `qsTr()` if this has been solved.

Reply:
Oh, nice to know it! I will update then it to `qsTr()`